skimage/color/colorconv.py

```diff
@@ -114,18 +126,22 @@ def convert_colorspace(arr, fromspace, tospace):
         msg = '`tospace` has to be one of {}'.format(todict.keys())
         raise ValueError(msg)
 
-    return todict[tospace](fromdict[fromspace](arr))
+    return todict[tospace](
+        fromdict[fromspace](arr, channel_axis=channel_axis),
+        channel_axis=channel_axis
+    )
 
 
-def _prepare_colorarray(arr, force_copy=False):
+def _prepare_colorarray(arr, force_copy=False, *, channel_axis=-1):
     """Check the shape of the array and convert it to
     floating point representation.
     """
     arr = np.asanyarray(arr)
 
-    if arr.shape[-1] != 3:
-        raise ValueError("Input array must have a shape == (..., 3)), "
-                         f"got {arr.shape}")
+    if arr.shape[channel_axis] != 3:
+        msg = ("the input array must have shape 3 along `channel_axis`, "
```

Comment:
```suggestion
        msg = (f"the input array must have size 3 along `channel_axis`, "
```
as in https://github.com/scikit-image/scikit-image/pull/5462/files#diff-3b92ef8526333aa6b6c52d8971304d93c4470d4ac47591fefe703145a43d6277R54?
shape would be like (3,) :wink: